packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -334,9 +339,10 @@ export const toStyles = ( tree, blockSelectors ) => {
 	const nodesWithSettings = getNodesWithSettings( tree, blockSelectors );
 
 	let ruleset =
-		'.wp-site-blocks > * { margin-top: 0; margin-bottom: 0; }.wp-site-blocks > * + * { margin-top: var( --wp--style--block-gap ); }';
+		'body { padding-right: 0; padding-left: 0; } .wp-site-blocks > * { margin-top: 0; margin-bottom: 0; padding-right: var(--wp--style--padding-right); padding-left: var(--wp--style--padding-left); }.wp-site-blocks > * + * { margin-top: var( --wp--style--block-gap ); }';
 	nodesWithStyles.forEach( ( { selector, styles } ) => {
-		const declarations = getStylesDeclarations( styles );
+		const isRoot = ROOT_BLOCK_SELECTOR === selector;
+		const declarations = getStylesDeclarations( styles, isRoot );
```

Comment:
I'm doing something similar in the backend over at https://github.com/WordPress/gutenberg/pull/39870/files#diff-c220112e642d42482d91af8a96dca5ab587adfa705140148f86c064bca995631R184, but passing the whole selector.